test/parallel/test-util-inspect.js

```diff
@@ -85,6 +85,88 @@ assert.strictEqual(util.inspect(async () => {}), '[AsyncFunction (anonymous)]');
   );
 }
 
+// Null constructor scenarios
+{
+  function fnNull() {}
+  Object.setPrototypeOf(fnNull, null);
+  assert.strictEqual(
+    util.inspect(fnNull),
+    '[Function (null prototype): fnNull]'
+  );
+
+  function fnNullAndStringTag() {}
+  Object.defineProperty(fnNullAndStringTag, Symbol.toStringTag, {
+    value: 'CustomTag',
+    configurable: true
+  });
+  Object.setPrototypeOf(fnNullAndStringTag, null);
+  assert.strictEqual(
+    util.inspect(fnNullAndStringTag),
+    '[Function (null prototype): fnNullAndStringTag] [CustomTag]'
+  );
+
+  function fnAnonymous() {}
+  Object.defineProperty(fnAnonymous, Symbol.toStringTag, {
+    value: 'AnonymousCustom',
+    configurable: true
+  });
+  Object.setPrototypeOf(fnAnonymous, null);
+  assert.strictEqual(
+    util.inspect(fnAnonymous),
+    '[Function (null prototype): fnAnonymous] [AnonymousCustom]'
+  );
+
+  const fnArrow = () => {};
+  Object.setPrototypeOf(fnArrow, null);
+  assert.strictEqual(
+    util.inspect(fnArrow),
+    '[Function (null prototype): fnArrow]'
+  );
+
+  async function fnAsync() {}
+  Object.defineProperty(fnAsync, Symbol.toStringTag, {
+    value: 'AsyncCustom',
+    configurable: true
+  });
+  Object.setPrototypeOf(fnAsync, null);
+  assert.strictEqual(
+    util.inspect(fnAsync),
+    '[AsyncFunction (null prototype): fnAsync] [AsyncCustom]'
+  );
+
+  class TestClass {}
+  Object.defineProperty(TestClass, Symbol.toStringTag, {
+    value: 'ClassTag',
+    configurable: true
+  });
+  Object.setPrototypeOf(TestClass, null);
+  assert.strictEqual(
+    util.inspect(TestClass),
+    '[class TestClass [ClassTag] extends [null prototype]]'
+  );
+
+  function fnMatchConstructor() {}
+  Object.defineProperty(fnMatchConstructor, Symbol.toStringTag, {
+    value: 'Function',
+    configurable: true
+  });
+  assert.strictEqual(
+    util.inspect(fnMatchConstructor),
+    '[Function: fnMatchConstructor]'
+  );
+
+  function fnNullMatchConstructor() {}
+  Object.defineProperty(fnNullMatchConstructor, Symbol.toStringTag, {
+    value: 'Function',
+    configurable: true
+  });
+  Object.setPrototypeOf(fnNullMatchConstructor, null);
+  assert.strictEqual(
+    util.inspect(fnNullMatchConstructor),
+    '[Function (null prototype): fnNullMatchConstructor] [Function]'
+  );
+}
```

Comment:
I believe these mostly duplicate existing tests. I am not sure if we really need these.

Reply:
hey, we actually have some cases testing these constructors, but I had added these null cases  because of the previous version that checked by `constructor === null`, I can remove them if you think they are unnecessary